ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2

```diff
@@ -76,21 +76,45 @@ port_channel_interfaces:
 {%                                     set short_esi_hash = (esi_hash | regex_replace("([0-9a-f]{4})","\\1:"))[:14] %}
-{%                                     do adapter_settings.port_channel.update({ 'short_esi': short_esi_hash }) %}
+{%                                     do adapter_settings.ethernet_segment.update({ 'short_esi': short_esi_hash }) %}
 {%                                 endif %}
-{%                                 if adapter_settings.port_channel.short_esi.split(':') | length == 3 %}
+{%                                 if adapter_settings.ethernet_segment.short_esi.split(':') | length == 3 %}
     evpn_ethernet_segment:
-      identifier: {{ adapter_settings.port_channel.short_esi | arista.avd.generate_esi(evpn_short_esi_prefix) }}
-      route_target: {{ adapter_settings.port_channel.short_esi | arista.avd.generate_route_target }}
+      identifier: {{ adapter_settings.ethernet_segment.short_esi | arista.avd.generate_esi(evpn_short_esi_prefix) }}
+{%                                     if adapter_settings.ethernet_segment.redundancy is arista.avd.defined %}
+      redundancy: {{ adapter_settings.ethernet_segment.redundancy }}
+{%                                     endif %}
+{%                                     if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined %}
+      designated_forwarder_election:
+{%                                         if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('modulus') %}
+        algorithm: modulus
+{%                                         elif adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('auto') %}
+{%                                             set auto_preferences = range(((adapter_settings.switches | length) * 100) - 100, 0, -100) %}
+        algorithm: preference
+        preference_value: {{ auto_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
+{%                                         elif adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined(var_type='list') %}
+        algorithm: preference
+        preference_value: {{ adapter_settings.ethernet_segment.designated_forwarder_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
+{%                                         endif %}
+{%                                         if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined and
+                                              adapter_settings.ethernet_segment.designated_forwarder_preferences is not arista.avd.defined('modulus') and
+                                              adapter_settings.ethernet_segment.dont_preempt is arista.avd.defined(true) %}
+        dont_preempt: true
+{%                                         endif %}
+{%                                     endif %}
```

Comment:
Same as for `ethernet-interface`.

line 97 you have already checked if the `designated_forwarder_precedence` is defined. so we could rewrite it like this (and I feel it could maybe even be simplified further but I am not finding how right now):

```suggestion
{%                                     if adapter_settings.ethernet_segment.redundancy is arista.avd.defined %}
      redundancy: {{ adapter_settings.ethernet_segment.redundancy }}
{%                                     endif %}
{%                                     if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined %}
      designated_forwarder_election:
{%                                         if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('modulus') %}
        algorithm: modulus
{%                                         elif adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('auto') %}
{%                                             set auto_preferences = range(((adapter_settings.switches | length) * 100) - 100, 0, -100) %}
        algorithm: preference
        preference_value: {{ auto_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
{%                                             if adapter_settings.ethernet_segment.dont_preempt is arista.avd.defined(true) %}
        dont_preempt: true
{%                                             endif %}
{%                                         elif adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined(var_type='list') %}
        algorithm: preference
        preference_value: {{ adapter_settings.ethernet_segment.designated_forwarder_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
{%                                             if adapter_settings.ethernet_segment.dont_preempt is arista.avd.defined(true) %}
        dont_preempt: true
{%                                             endif %}
{%                                         endif %}
{%                                     endif %}
```

Reply:
Thanks for the suggestions.  I've refactored the logic because of @ClausHolbechArista 's schema recommendations and tried to get down to fewer levels of nesting and compound conditionals.